internal/armada/queryapi/database/query.sql

```diff
@@ -1,2 +1,12 @@
--- name: GetJobState :many
-SELECT state FROM job WHERE job_id = $1;
+-- name: GetJobStates :many
+SELECT job_id, state FROM job WHERE job_id = ANY(sqlc.arg(job_ids)::text[]);
+
+-- name: GetJobDetails :many
+SELECT job_id, queue, jobset, namespace, state, submitted, cancelled, cancel_reason, last_transition_time, latest_run_id, job_spec FROM job WHERE job_id = ANY(sqlc.arg(job_ids)::text[]);
+
+-- name: GetJobRunsByRunIds :many
+SELECT * FROM job_run WHERE run_id = ANY(sqlc.arg(run_ids)::text[]);
+
+-- name: GetJobRunsByJobIds :many
+SELECT * FROM job_run WHERE job_id = ANY(sqlc.arg(job_ids)::text[]) order by leased  desc;
```

Comment:
Does this need the order by?

Reply:
Yes- two reasons:
1.  It means that the job runs appear in chronological order (i.e. the active run will be first)
2. It makes unit testing much easier becuase you have a deterministic result.

The performance overhead of this *should* be be negligable unless we are retrieving a very large number of runs.